Registering an XQuery module with a syntax error does not point to the position of the error

Registering an XQuery module with a syntax error does not point to the position of where this error is found. This does work when parsing an XPath query.

Try, for example, to register the following XQuery module:

```xquery
declare %public function local:test () as xs:string {
	let $error := "not-so-obvious error ->",
	return $error
}
```

Registering that module will result in the following error message:

```
AssertionError: Unable to load all files: SyntaxError: Expected " ", "$", "(:", "\n", "\r", or "\t" but "r" found.
    at Context.eval (<webpack>)
```

--- a/src/registerXQueryModule.ts
+++ b/src/registerXQueryModule.ts
@@ -1,20 +1,25 @@
 import ExecutionSpecificStaticContext from './expressions/ExecutionSpecificStaticContext';
 import StaticContext from './expressions/StaticContext';
 import astHelper from './parsing/astHelper';
-import { parse } from './parsing/xPathParser';
-
 import { loadModuleFile } from './parsing/globalModuleCache';
-
+import parseExpression from './parsing/parseExpression';
 import processProlog from './parsing/processProlog';
 
 /**
  * Register an XQuery module
  * @public
  * @param   moduleString - The string contents of the module
+ * @param   options - Additional compilation options
  * @returns  The namespace uri of the new module
  */
-export default function registerXQueryModule(moduleString: string): string {
-	const parsedModule = parse(moduleString, { outputDebugInfo: false, xquery: true });
+export default function registerXQueryModule(
+	moduleString: string,
+	options: { debug: boolean } = { debug: false }
+): string {
+	const parsedModule = parseExpression(moduleString, {
+		debug: options['debug'],
+		allowXQuery: true
+	});
 
 	const libraryModule = astHelper.getFirstChild(parsedModule, 'libraryModule');
 	if (!libraryModule) {

